src/python/WMCore/WMSpec/Steps/Executors/LogArchive.py

```diff
@@ -69,76 +65,79 @@ def execute(self, emulator = None, **overrides):
 
         logging.info("Beginning Steps.Executors.LogArchive.Execute")
-        logging.info("Using the following overrides: %s " % overrides)
-        logging.info("Step is: %s" % self.step)
-        # Wait timout for stageOut
+        logging.info("Using the following overrides: %s ", overrides)
+        logging.info("Step is: %s", self.step)
+        # Wait timeout for stageOut
         waitTime = overrides.get('waitTime', 3600 + (self.step.retryDelay * self.step.retryCount))
 
         matchFiles = [
-            ".log$",
-            "FrameworkJobReport",
+            ".log$",  # matches the scram, wmagent and cmsRun logs
+            "FrameworkJobReport.xml",
             "Report.pkl",
-            "Report.pcl",
             "^PSet.py$",
-            "^PSet.pkl$"
-            ]
+            "^PSet.pkl$",
+            "_condor_std*",  # condor wrapper logs at the pilot top level
+        ]
+        ignoredDirs = ['Utils', 'WMCore', 'WMSandbox']
 
-        #Okay, we need a stageOut Manager
+        # Okay, we need a stageOut Manager
         useNewStageOutCode = False
         if getattr(self.step, 'newStageout', False) or \
-            ('newStageOut' in overrides and overrides.get('newStageOut')):
+                ('newStageOut' in overrides and overrides.get('newStageOut')):
             useNewStageOutCode = True
         if not useNewStageOutCode:
             # old style
             manager = StageOutMgr.StageOutMgr(**overrides)
             manager.numberOfRetries = self.step.retryCount
-            manager.retryPauseTime  = self.step.retryDelay
+            manager.retryPauseTime = self.step.retryDelay
         else:
             # new style
             logging.info("LOGARCHIVE IS USING NEW STAGEOUT CODE")
             manager = WMCore.Storage.FileManager.StageOutMgr(
-                                retryPauseTime  = self.step.retryDelay,
-                                numberOfRetries = self.step.retryCount,
-                                **overrides)
-
-        #Now we need to find all the reports
-        logFilesForTransfer = []
-        #Look in the taskSpace first
-        logFilesForTransfer.extend(self.findFilesInDirectory(self.stepSpace.taskSpace.location, matchFiles))
-
-        #What if it's empty?
+                retryPauseTime=self.step.retryDelay,
+                numberOfRetries=self.step.retryCount,
+                **overrides)
+
+        # Now we need to find all the reports
+        # The log search follows this structure: ~pilotArea/jobArea/WMTaskSpaceArea/StepsArea
+        # Start looking at the pilot scratch area first, such that we find the condor logs
+        # Then look at the job area in order to find the wmagentJob log
+        # Finally, at the taskspace area to find the cmsRun/FWJR/PSet files
+        pilotScratchDir = os.path.join(self.stepSpace.taskSpace.location, '../../')
+        logFilesForTransfer = self.findFilesInDirectory(pilotScratchDir, matchFiles, ignoredDirs)
```

Comment:
logFilesForTransfe - Is that the files archived?

Reply:
list of files to be archived in the tarball. Let me update it.